localization/ekf_localizer/src/ekf_localizer.cpp

```diff
@@ -111,13 +112,57 @@ EKFLocalizer::EKFLocalizer(const std::string & node_name, const rclcpp::NodeOpti
   pub_measured_pose_ = create_publisher<geometry_msgs::msg::PoseStamped>("debug/measured_pose", 1);
 }
 
+/*
+ * calcTimerRate
+ */
+double EKFLocalizer::calcTimerRate(size_t window_size) const
+{
+  if (time_buffer_.size() < window_size) {
+    return ekf_rate_;
+  }
+
+  const auto time_diff = (time_buffer_.back() - time_buffer_.front()).seconds();
+  const auto num_intervals = time_buffer_.size() - 1;
+
+  return static_cast<double>(num_intervals) / time_diff;
+}
+
+/*
+ * updatePredictFrequency
+ */
+void EKFLocalizer::updatePredictFrequency()
+{
+  size_t window_size = 10;
+  time_buffer_.push_back(clock_->now());
+  while (static_cast<int>(time_buffer_.size()) > window_size) {
+    time_buffer_.pop_front();
+  }
+
+  // Calc timer rate
+  double timer_rate = calcTimerRate(window_size);
+  if (timer_rate < ekf_rate_ * 0.99 || timer_rate > ekf_rate_ * 1.01) {
+    ekf_rate_ = timer_rate;
```

Comment:
@YamatoAndo @TakaHoribe 
Should `ekf_rate` be smaller than clock rate like `ekf_rate_ = timer_rate/2;`?

Reply:
sorry, If doing this, the predict rate and ekf_dt will not match again.